lib/analyzer/applications/runtime-common.ts

```diff
@@ -1,23 +1,82 @@
 import * as path from "path";
+import { parsePkgJson } from "snyk-nodejs-lockfile-parser";
 import { ApplicationFilesFact } from "../../facts";
 import {
   AppDepsScanResultWithoutTarget,
   ApplicationFileInfo,
   FilePathToContent,
 } from "./types";
 
-export function getAppFilesRootDir(
-  filePaths: string[],
-): [string, ApplicationFileInfo[]] {
+interface AppFileMetadataExtractor {
+  manifestFiles: string[];
```

Comment:
1. I would changing it to 'fileMatcher'
2. Are there consts that can be used instead of the hard-coded package.json below?
3. I would change Record<string, string> to something more specific (an interface that describe the metadata returned from package.json